client/src/legacy/GridField.js

```diff
@@ -200,42 +200,69 @@ $.entwine('ss', function($) {
       }
     },
 
+    /**
+     * This will first retrieve state of the gridfield which is generated by PHP and sent to the
+     * browser as HTML via PJAX and stored in the data-schema attribute on all of the '3 dots'
+     * elements on each gridfield row
+     * 
+     * It will then update the browser location with the new state of the gridfield using
+     * window.ss.router
+     * 
+     * This allows users to bookmark different views in the CMS
+     */
     keepStateInHistory: function() {
       const newSchema = $(this).find('.gridfield-actionmenu__container').data('schema');
+      const gridFieldName = $(this).data('name');
       if (newSchema && newSchema.length > 0) {
         newSchema.filter( e => {
           if (e.type === 'link') {
-            const reqString = this.buildURLString(e.url)
-            const searchParam =  reqString ? '?' + reqString.join('&') : '';
-            window.ss.router.replace(window.location.pathname + searchParam, undefined, undefined, false);
+            const urlQueryString = this.buildUrlQueryString(e.url, gridFieldName);
```

Comment:
Probably, we need to encode request string as before.

Reply:
It wasn't being encoded before? I can't see anything I've deleted that has to do with encoding? Am I missing something?

Reply:
`new URL(link)).searchParams.toString()` returns encoded URL.

Reply:
new URL(link)).searchParams.toString() returns encoded URL.

Reply:
We do that on line 251

Reply:
It doesn't work, because `toString()` does all magic at the end.

Reply:
That's an improvement then?  Seems like the browser works fine with it encoded or not so we may as well not encode it?

Reply:
Ah right I see. OK I've added in `encodeURI()`